cloud/azure/azure_rm_networkinterface.py

```diff
@@ -0,0 +1,619 @@
+'''
+
+EXAMPLES = '''
+    - name: Create a network interface with minimal parameters
+        azure_rm_networkinterface:
+            name: nic001
+            resource_group: Testing
+            virtual_network_name: vnet001
+            subnet_name: subnet001
+
+     - name: Create a network interface with private IP address only (no Public IP)
+        azure_rm_networkinterface:
+            name: nic001
+            resource_group: Testing
+            virtual_network_name: vnet001
+            subnet_name: subnet001
+            public_ip: no
+
+    - name: Create a network interface for use in a Windows host (opens RDP port) with custom RDP port
+        azure_rm_networkinterface:
+            name: nic002
+            resource_group: Testing
+            virtual_network_name: vnet001
+            subnet_name: subnet001
+            os_type: Windows
+            rdp_port: 3399
+
+    - name: Create a network interface using existing security group and public IP
+        azure_rm_networkinterface:
+            name: nic003
+            resource_group: Testing
+            virtual_network_name: vnet001
+            subnet_name: subnet001
+            security_group_name: secgroup001
+            public_ip_address_name: publicip001
+
+    - name: Delete network interface
+        azure_rm_networkinterface:
+            resource_group: Testing
+            name: nic003
+            state: absent
+'''
+
+RETURN = '''
+changed:
+    description: Whether or not the object was changed.
+    returned: always
+    type: bool
+    sample: True
+check_mode:
+    description: Whether or not the module was executed in check mode.
+    returned: always
+    type: bool
+    sample: True
+Results:
+    description: Facts about the current state of the object.
+    returned: always
+    type: dict
+    sample: {
+        "dns_settings": {
+            "applied_dns_servers": [],
+            "dns_servers": [],
+            "internal_dns_name_label": null,
+            "internal_fqdn": null
+        },
+        "enable_ip_forwarding": false,
+        "etag": "W/\"be115a43-2148-4545-a324-f33ad444c926\"",
+        "id": "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/Testing/providers/Microsoft.Network/networkInterfaces/nic003",
+        "ip_configuration": {
+            "name": "default",
+            "private_ip_address": "10.1.0.10",
+            "private_ip_allocation_method": "Static",
+            "public_ip_address": {
+                "id": "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/Testing/providers/Microsoft.Network/publicIPAddresses/publicip001",
+                "name": "publicip001"
+            },
+            "subnet": {}
+        },
+        "location": "eastus2",
+        "mac_address": null,
+        "name": "nic003",
+        "network_security_group": {},
+        "primary": null,
+        "provisioning_state": "Succeeded",
+        "tags": null,
+        "type": "Microsoft.Network/networkInterfaces"
```

Comment:
since these are 'facts' it might be worth returning ansbile_facts.<azure_namespace>.<interface_namespace>?
or just udate the ansible_interfaces facts?